### DISCUSSION TITLE
Collection::splice return empty

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#52252] [General]
```php
$collection = collect([1, 2, 3]);

$collection->splice(1, 0, 4); <------ returns empty collection

dd($collection->all()); <------ return [1,4,2,3]
```

What if i want to use fluent syntax after calling `splice` method ?

**Comment by noefleury** [comment]
From what I see you want to insert some element `4` at the index `1` of the collection ?

Not sure the [splice method was made](https://laravel.com/docs/11.x/collections#method-splice) to be used fluently for this type of case.

And a proposal about an insert in Collection was rejected : https://github.com/laravel/ideas/issues/650

This code is a bit tricky but seems to work fluently (using variables for better understanding) :

```php
$collection = collect([1, 2, 3]);
$insertionIndex = 1;
$insertionValue = 4;
$collection->slice(0, $insertionIndex)->push($insertionValue)->merge($collection->splice($insertionIndex))->dump();
```
